### DISCUSSION TITLE
cloudflare

### DISCUSSION THREAD

**Discussion by strangelion** [jeessy2/ddns-go#1159] [Q&A] [answered]
为什么这个一直在尝试创建新的解析，导致每次更换ipv6后我都要把之前的删除
![Uploading Snipaste_2024-06-16_08-32-22.png…]()

**Comment by strangelion** [comment]
我用ali的正常更换，但是cloudflare的每次都创建新的

**Comment by jeessy2** [comment]
token的权限给够了吗？

**Comment by strangelion** [comment]
给了编辑dns的权限

**Reply by jeessy2** [reply]
多试下吧，肯定是可以的

**Comment by strangelion** [comment]
现在情况是阿里云的解析每次都可以正常更换，然而cloudflare的每次都是创建一个新的解析，导致我每次都要把旧的删除

**Reply by jeessy2** [reply]
重新创建token

**Reply by strangelion** [reply]
还是不行，又创建了新的解析

**Reply by jeessy2** [reply]
把你配置，如何创建的发来。只说一个不行/还是不行，这个无法看

**Comment by strangelion** [comment]
- name: ""
      ipv4:
        enable: false
        gettype: url
        url: https://myip.ipip.net, https://ddns.oray.com/checkip, https://ip.3322.net, https://4.ipw.cn
        netinterface: WLAN 5
        cmd: ""
        domains:
            - ""
      ipv6:
        enable: true
        gettype: url
        url: https://speed.neu6.edu.cn/getIP.php, https://v6.ident.me, https://6.ipw.cn
        netinterface: WLAN 5
        cmd: ""
        ipv6reg: ""
        domains:
            - pan:(这里是我的域名)?proxied=true
            - blog:(这里是我的域名)?proxied=true
            - www:(这里是我的域名)?proxied=true
            - minecraft:(这里是我的域名)?proxied=true
      dns:
        name: cloudflare
        id: ""
        secret:这里又写了我的api
      ttl: ""

Api创建方法:创建令牌-编辑区域dns(使用模板),去选择那里选择了要解析的地址-选择继续以显示摘要
________________________________
发件人: jeessy2 ***@***.***>
发送时间: 星期日, 六月 16, 2024 8:53:42 上午
收件人: jeessy2/ddns-go ***@***.***>
抄送: strangelion ***@***.***>; Author ***@***.***>
主题: Re: [jeessy2/ddns-go] cloudflare (Discussion #1159)


把你配置，如何创建的发来。只说一个不行/还是不行，这个无法看

―

**Reply by jeessy2** [reply]
日志发一个

**Comment by strangelion** [comment]
2024/06/16 08:48:39 新增域名解析 pan.（这里是我的域名） 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:48:40 新增域名解析 blog.（这里是我的域名） 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:48:41 新增域名解析 www.（这里是我的域名） 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:48:42 新增域名解析 minecraft.（这里是我的域名） 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:49:09 IPv6未改变, 将等待 5 次后与DNS服务商进行比对
2024/06/16 08:49:11 新增域名解析 pan.（这里是我的域名） 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:49:12 新增域名解析 blog.（这里是我的域名） 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:49:13 新增域名解析 www.（这里是我的域名） 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:49:14 新增域名解析 minecraft.（这里是我的域名） 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:54:14 IPv6未改变, 将等待 4 次后与DNS服务商进行比对
2024/06/16 08:54:16 新增域名解析 pan.（这里是我的域名） 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:54:17 新增域名解析 blog.（这里是我的域名） 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:54:18 新增域名解析 www.（这里是我的域名） 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:54:19 新增域名解析 minecraft.（这里是我的域名） 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:59:19 IPv6未改变, 将等待 3 次后与DNS服务商进行比对
2024/06/16 08:59:21 新增域名解析 pan.（这里是我的域名） 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:59:22 新增域名解析 blog.（这里是我的域名） 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:59:23 新增域名解析 www.（这里是我的域名） 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:59:24 新增域名解析 minecraft.（这里是我的域名） 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 09:04:24 IPv6未改变, 将等待 2 次后与DNS服务商进行比对
2024/06/16 09:04:26 新增域名解析 pan.（这里是我的域名） 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 09:04:27 新增域名解析 blog.（这里是我的域名） 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 09:04:28 新增域名解析 www.（这里是我的域名） 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 09:04:29 新增域名解析 minecraft.（这里是我的域名） 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400

________________________________
发件人: jeessy2 ***@***.***>
发送时间: 星期日, 六月 16, 2024 9:07:23 上午
收件人: jeessy2/ddns-go ***@***.***>
抄送: strangelion ***@***.***>; Author ***@***.***>
主题: Re: [jeessy2/ddns-go] cloudflare (Discussion #1159)


日志发一个

―

**Reply by jeessy2** [reply]
我的域名是什么内容，把关键信息隐藏

**Reply by strangelion** [reply]
2024/06/16 08:47:25 你的IP 这是ipv6（我隐藏一下） 没有变化, 域名 pan.（我的域名）.com
2024/06/16 08:47:25 你的IP 这是ipv6（我隐藏一下） 没有变化, 域名 blog.（我的域名）.com
2024/06/16 08:47:25 你的IP 这是ipv6（我隐藏一下） 没有变化, 域名 www.（我的域名）.com
2024/06/16 08:47:25 你的IP 这是ipv6（我隐藏一下） 没有变化, 域名 minecraft.（我的域名）.com
2024/06/16 08:47:26 新增域名解析 pan.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:47:27 新增域名解析 blog.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:47:28 新增域名解析 www.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:47:30 新增域名解析 minecraft.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:48:36 配置文件已保存在: C:\Users\王文轩\.ddns_go_config.yaml
2024/06/16 08:48:37 你的IP 这是ipv6（我隐藏一下） 没有变化, 域名 pan.（我的域名）.com
2024/06/16 08:48:37 你的IP 这是ipv6（我隐藏一下） 没有变化, 域名 blog.（我的域名）.com
2024/06/16 08:48:37 你的IP 这是ipv6（我隐藏一下） 没有变化, 域名 www.（我的域名）.com
2024/06/16 08:48:37 你的IP 这是ipv6（我隐藏一下） 没有变化, 域名 minecraft.（我的域名）.com
2024/06/16 08:48:39 新增域名解析 pan.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:48:40 新增域名解析 blog.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:48:41 新增域名解析 www.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:48:42 新增域名解析 minecraft.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:49:09 IPv6未改变, 将等待 5 次后与DNS服务商进行比对
2024/06/16 08:49:11 新增域名解析 pan.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:49:12 新增域名解析 blog.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:49:13 新增域名解析 www.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:49:14 新增域名解析 minecraft.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:54:14 IPv6未改变, 将等待 4 次后与DNS服务商进行比对
2024/06/16 08:54:16 新增域名解析 pan.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:54:17 新增域名解析 blog.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:54:18 新增域名解析 www.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:54:19 新增域名解析 minecraft.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:59:19 IPv6未改变, 将等待 3 次后与DNS服务商进行比对
2024/06/16 08:59:21 新增域名解析 pan.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:59:22 新增域名解析 blog.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:59:23 新增域名解析 www.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 08:59:24 新增域名解析 minecraft.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 09:04:24 IPv6未改变, 将等待 2 次后与DNS服务商进行比对
2024/06/16 09:04:26 新增域名解析 pan.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 09:04:27 新增域名解析 blog.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 09:04:28 新增域名解析 www.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 09:04:29 新增域名解析 minecraft.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 09:09:29 IPv6未改变, 将等待 1 次后与DNS服务商进行比对
2024/06/16 09:09:31 新增域名解析 pan.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 09:09:32 新增域名解析 blog.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 09:09:33 新增域名解析 www.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 09:09:34 新增域名解析 minecraft.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 09:14:35 你的IP 这是ipv6（我隐藏一下） 没有变化, 域名 pan.（我的域名）.com
2024/06/16 09:14:35 你的IP 这是ipv6（我隐藏一下） 没有变化, 域名 blog.（我的域名）.com
2024/06/16 09:14:35 你的IP 这是ipv6（我隐藏一下） 没有变化, 域名 www.（我的域名）.com
2024/06/16 09:14:35 你的IP 这是ipv6（我隐藏一下） 没有变化, 域名 minecraft.（我的域名）.com
2024/06/16 09:14:37 新增域名解析 pan.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 09:14:38 新增域名解析 blog.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 09:14:40 新增域名解析 www.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400
2024/06/16 09:14:41 新增域名解析 minecraft.（我的域名）.com 失败! 异常信息: 返回内容: {"result":null,"success":false,"errors":[{"code":81057,"message":"Record already exists."}],"messages":[]} ,返回状态码: 400

**Reply by strangelion** [reply]
dnsconf:
    - name: ""
      ipv4:
        enable: false
        gettype: url
        url: https://myip.ipip.net, https://ddns.oray.com/checkip, https://ip.3322.net, https://4.ipw.cn
        netinterface: WLAN 5
        cmd: ""
        domains:
            - ""
      ipv6:
        enable: true
        gettype: url
        url: https://speed.neu6.edu.cn/getIP.php, https://v6.ident.me, https://6.ipw.cn
        netinterface: WLAN 5
        cmd: ""
        ipv6reg: ""
        domains:
            - pan:（我的域名）.com
            - blog:（我的域名）.com
            - www:（我的域名）.com
            - minecraft:（我的域名）.com
      dns:
        name: alidns
        id: 阿里云id
        secret: 阿里云secret
      ttl: ""
    - name: ""
      ipv4:
        enable: false
        gettype: url
        url: https://myip.ipip.net, https://ddns.oray.com/checkip, https://ip.3322.net, https://4.ipw.cn
        netinterface: WLAN 5
        cmd: ""
        domains:
            - ""
      ipv6:
        enable: true
        gettype: url
        url: https://speed.neu6.edu.cn/getIP.php, https://v6.ident.me, https://6.ipw.cn
        netinterface: WLAN 5
        cmd: ""
        ipv6reg: ""
        domains:
            - pan:（我的域名）.com?proxied=true
            - blog:（我的域名）.com?proxied=true
            - www:（我的域名）.com?proxied=true
            - minecraft:（我的域名）.com
      dns:
        name: cloudflare
        id: ""
        secret: cloudfare的api
      ttl: ""
user:
    username: 这是登录的用户名
    password: 这是密码
webhook:
    webhookurl: ""
    webhookrequestbody: ""
    webhookheaders: ""
notallowwanaccess: true
lang: zh
这是完整的配置文件

**Reply by jeessy2** [reply]
哥，你创建了二个一样的配置。。。。才看到。。。

**Comment by jeessy2** [comment]
刚测了下。正常
你隐藏的域名是不是顶级域名，而不是免费的或谁送的？

**Comment by strangelion** [comment]
不，就是我买的一个域名，把中间那个隐藏而已，其实那个域名包含.com的。

________________________________
发件人: jeessy2 ***@***.***>
发送时间: 星期日, 六月 16, 2024 9:35:24 上午
收件人: jeessy2/ddns-go ***@***.***>
抄送: strangelion ***@***.***>; Author ***@***.***>
主题: Re: [jeessy2/ddns-go] cloudflare (Discussion #1159)


刚测了下。正常
你隐藏的域名是不是顶级域名，而不是免费的或谁送的？

―

**Comment by strangelion** [comment]
要删除一个吗？
因为我在阿里和cloudflare上都弄了一个解析

________________________________
发件人: jeessy2 ***@***.***>
发送时间: 星期日, 六月 16, 2024 9:48:26 上午
收件人: jeessy2/ddns-go ***@***.***>
抄送: strangelion ***@***.***>; Author ***@***.***>
主题: Re: [jeessy2/ddns-go] cloudflare (Discussion #1159)


哥，你创建了二个一样的配置。。。。才看到。。。

―

**Reply by jeessy2** [reply]
cloudflare添加了主域名没有？

**Comment by strangelion** [comment]
添加了，我域名的dns服务器这两天才添加了cloudflare[cid:6b0526ac-ad47-4343-8559-fddeed79326c]

________________________________
发件人: jeessy2 ***@***.***>
发送时间: 星期日, 六月 16, 2024 9:56:10 上午
收件人: jeessy2/ddns-go ***@***.***>
抄送: strangelion ***@***.***>; Author ***@***.***>
主题: Re: [jeessy2/ddns-go] cloudflare (Discussion #1159)


cloudflare添加了主域名没有？

―

**Reply by jeessy2** [reply]
中文域名？

**Comment by strangelion** [comment]
对，照片上应该有吧。

获取Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by jeessy2** [reply]
你照片没传起，看不到。中文域名不要用cf之类的，你还是用阿里云吧。。。

**Comment by strangelion** [comment]
行吧

获取Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by strangelion** [comment]
发现方法了，我把中文的域名变成英文的（xn开头的，浏览器复制的时候可以复制出来）就正常了
________________________________

**Reply by jeessy2** [reply]
也可以，不过这样不好看

**Reply by jeessy2** [reply, marked as answer]
Fix by @WaterLemons2k
Try: https://github.com/jeessy2/ddns-go/releases/tag/v6.6.3
用`😺.你的域名.com`测下并反馈

**Comment by strangelion** [comment]
目前测试是能正常换ip v6
________________________________
